core/types/block_test.go

```diff
@@ -67,6 +70,126 @@ func TestBlockEncoding(t *testing.T) {
 	}
 }
 
+// Test BLS transaction type block encoding/decoding.
+func TestEIP7591BlockEncoding(t *testing.T) {
+	// Fields to build block
+	var (
+		txs      = make([]*Transaction, 2)
+		receipts = make([]*Receipt, len(txs))
+		uncles   = make([]*Header, 0)
+	)
+
+	// Generate BLS key
+	k, err := newKey()
+	if err != nil {
+		t.Fatal("failed to generate BLS keys:", err)
+	}
+
+	check := func(f string, got, want interface{}) {
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("%s mismatch: got %v, want %v", f, got, want)
+		}
+	}
+
+	// Sign and aggregate a BLS signature
+	msg := make([]byte, 50)
+	sig := bls.Sign(k.sk, msg)
+	aggSig, err := bls.AggregateSignatures([]*bls.Signature{sig})
+	if err != nil {
+		t.Fatal("failed to aggregate BLS signatures:", err)
+	}
+	check("Aggregated Signature", sig, aggSig)
+
+	// Edit the header fields to include BLS AggregatedSig field
+	header := &Header{
+		Difficulty:       big.NewInt(285311670611),
+		Number:           math.BigPow(2, 9),
+		GasLimit:         12345678,
+		GasUsed:          1476322,
+		Time:             9876543,
+		Extra:            []byte("coolest block on chain"),
+		WithdrawalsHash:  &EmptyWithdrawalsHash,
+		ParentBeaconRoot: new(common.Hash),
+		AggregatedSig:    bls.SignatureToBytes(aggSig),
```

Comment:
This test will change after the Block Building changes are made. 

The Signature Aggregation will happen when building a `NewBlock`